src/monoflow.ts

```diff
@@ -17,47 +18,58 @@ export class Workflow<Z, A, B> {
     private readonly _steps: readonly Workflow<any, any, any>[] = []
   ) {}
 
-  then<C>(fn: (item: B) => C): Workflow<Z, B, C> {
+  then<C>(fn: (item: B) => C): Workflow<Z, B, C> | Workflow<Z, A, C>{
+    // Make a higher-order function when `this._err` exists.
+    if (this._ok) {
+      const binded = this._ok.bind(this);
+      return new Workflow((item: A) => fn(binded(item)), undefined, [...this._steps]);
+    }
     return new Workflow(fn, undefined, [...this._steps, this]);
   }
 
-  else<C>(fn: (error: Error) => C): Workflow<Z, B, C> {
+  else<C>(fn: (error: Error) => C): Workflow<Z, B, B | C>  {
+    // Make a higher-order function when `this._err` exists.
+    if (this._err) {
+      console.log("here");
+      const binded = this._err.bind(this);
+      const safeFn = (err: Error) => {
+        try {
+          return binded(err);
+        } catch (err2) {
+          return err2;
+        }
+      }
+      return new Workflow(undefined, (error: Error) => fn(safeFn(error) as Error), [...this._steps]);
```

Comment:
This seems incorrect to me. `fn` should only be called with `Error`, not `B`

Reply:
```ts
  const workflow = Workflow.create((n: number) => {
    throw new Error("hey");
  })
    .else((_err) => {
      return "hey";
    })
    .else((err) => {
      // oops! `err` is actually a string, not Error type :(
      return `${err.message}!`;
    });
  expect(workflow.run(1)).toBe("hey!");
```

Reply:
OK, I fixed to ignore the second one if the first one returns value correctly.